app/controllers/onboarding_controller.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+class OnboardingController < ApplicationController
+  skip_before_action :authenticate
+  before_action :verify_token
+
+  layout 'onboarding'
+
+  def index
+    @token = token_param
+    @user = User.new
+  end
+
+  def create
+    # Ensure information on registered users is never
+    # disclosed during onboarding
+    @user = User.find_by(email: user_params[:email])
```

Comment:
That’s correct – we’re using the `valid_email2` gem for email validation, which – as far as I can see – doesn’t have a helper to get some kind of normalised email, right? So I guess the best option would be to instantiate a new user and try to call the validator? Or do you know of a better solution?

Reply:
Encoding the user's id in the token would fix this problem, users could enter any email they want and then they would update the user account instead of creating a new one.

Reply:
User do not exist at this point – the whole idea is to be able to simply pass on a link to all contributors via a bcc email, a direct chat message, … so that they can self-register.

Reply:
So to rephrase my question: What would be the most reliable way to detect when rails would otherwise display a "email not unique" error. (Does `valid_email2` even have any impact on this behaviour?)

The purpose of this check is not to prevent users from registering two times with slightly different versions, it is not to disclose if a user has already registered.

Reply:
I just pushed a commit that changes the behavior: The controller now relies on the model validators to check if an email address is already taken.